apps/native/app/src/screens/document-detail/document-communications.tsx

```diff
@@ -0,0 +1,283 @@
+import {
+  Alert,
+  Button,
+  Container,
+  ListItemSkeleton,
+  TopLine,
+  Typography,
+} from '../../ui'
+import { ComponentRegistry } from '../../utils/component-registry'
+import { createSkeletonArr } from '../../utils/create-skeleton-arr'
+import {
+  DocumentListItem,
+  TOGGLE_ANIMATION_DURATION,
+} from './components/document-list-item'
+import {
+  FloatingBottomContent,
+  FloatingBottomFooter,
+} from './components/floating-bottom-footer'
+import { DocumentReplyScreenProps } from './document-reply'
+
+type FlatListItem = DocumentComment | { __typename: 'Skeleton'; id: string }
+
+const LAST_REPLY_HEIGHT = 100
+
+const CaseNumberWrapper = styled.View`
+  flex-direction: row;
+  align-items: center;
+  justify-content: center;
+  column-gap: ${({ theme }) => theme.spacing.smallGutter}px;
+`
+
+const ListWrapper = styled.View`
+  flex: 1;
+  padding-top: ${({ theme }) => theme.spacing[2]}px;
+`
+
+const { useNavigationOptions, getNavigationOptions } =
+  createNavigationOptionHooks(
+    () => ({
+      topBar: {
+        noBorder: true,
+      },
+    }),
+    {
+      bottomTabs: {
+        visible: false,
+      },
+    },
+  )
+
+export type DocumentCommunicationsScreenProps = {
+  documentId: string
+  ticketId?: string
+}
+
+export const DocumentCommunicationsScreen: NavigationFunctionComponent<
+  DocumentCommunicationsScreenProps
+> = ({ componentId, documentId, ticketId }) => {
+  useNavigationOptions(componentId)
+  const locale = useLocale()
+  const theme = useTheme()
+  const intl = useIntl()
+  const formatDate = useDateTimeFormatter()
+  const { showModal } = useNavigationModal()
+  const { user } = useUser()
+  const insets = useSafeAreaInsets()
+  const [shouldScroll, setShouldScroll] = useState(false)
+  const [buttonHeight, setButtonHeight] = useState(48) // Default height
+  const [refetching, setRefetching] = useState(false)
+  const [showFirstReplyInfo, setShowFirstReplyInfo] = useState(false)
+  const flatListRef = useRef<Animated.FlatList>(null)
+  const scrollY = useRef(new Animated.Value(0)).current
+
+  const docRes = useGetDocumentQuery({
+    variables: {
+      input: {
+        id: documentId,
+      },
+      locale,
+    },
+  })
+
+  useConnectivityIndicator({
+    componentId,
+    queryResult: docRes,
+    refetching,
+  })
+
+  const document = docRes.data?.documentV2
+  const comments = document?.ticket?.comments ?? []
+  const replyable =
+    !docRes.loading && document && document?.replyable
+      ? document?.replyable
+      : true
+  const isSkeleton = docRes.loading && !docRes.data
+
+  const handleRefresh = () => {
+    setRefetching(true)
+    docRes.refetch().finally(() => setRefetching(false))
+  }
+
+  const onReplyPress = () => {
+    if (!document?.sender?.name) {
+      return
+    }
+
+    const passProps: DocumentReplyScreenProps = {
+      senderName: document.sender.name,
+      documentId,
+      subject: document?.subject,
+      onReplySuccess: (showFirstReplyInfo) => {
```

Comment:
Passing props when closing native modals in  React native navigation is not possible. A callback prop however works rather nice and better than having context or store in this instance